src/url_builder.rs

```diff
@@ -99,11 +105,28 @@ impl UrlBuilder {
         self
     }
 
+    pub fn alternates(&mut self, alternates: Vec<Alternate>) -> &mut Self {
+        self.alternates = Some(alternates);
+        self
+    }
+
+    pub fn push_alternate(&mut self, hreflang: String, href: String) -> &mut Self {
```

Comment:
Can you use `impl ToString` instead of `String` for `hreflang` and `href` here? This makes for a nicer interface.